### PR TITLE
WIP: Normalize similarities to [0, 1]

### PR DESCRIPTION
**Pull request by edwinyyyu**
### Purpose of the change

Define common similarity/distance definitions.

Score threshold is currently defined as bigger is better, and the parameter is supposed to be overloaded instead of creating new parameters for different systems.

Supposedly users may be unwilling to understand the raw values.

### Description

Define score formulas.

TODO:
- Implement score formulas for different database backends.
- Consider impacts on rerankers. Some rerankers return range [0, 1], [-1, 1], (-inf, inf) e.g. logits, etc.

Alternative:
- define different scores for different systems and return raw values (easiest to understand)

Breaking:
- existing tuned score thresholds no longer work if used for embedding score

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (does not change functionality, e.g., code style improvements, linting)
- [ ] Documentation update
- [ ] Project Maintenance (updates to build scripts, CI, etc., that do not affect the main project)
- [ ] Security (improves security without changing functionality)

### How Has This Been Tested?

- [ ] Unit Test
- [ ] Integration Test
- [ ] End-to-end Test
- [ ] Test Script (please provide)
- [ ] Manual verification (list step-by-step instructions)

### Checklist

- [ ] I have signed the commit(s) within this pull request
- [ ] My code follows the style guidelines of this project (See STYLE_GUIDE.md)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### Maintainer Checklist

- [ ] Confirmed all checks passed
- [ ] Contributor has signed the commit(s)
- [ ] Reviewed the code
- [ ] Run, Tested, and Verified the change(s) work as expected

### CODE REVIEW

**Comment by edwinyyyu** [comment]
I don't like these definitions since they are so complicated and it's a lot of effort to use if someone is already familiar with their embedding model and have a threshold in mind.

But same direction for better score makes a lot of the consumer logic much simpler.
